x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/type/EsqlDataTypeConverter.java

```diff
@@ -370,7 +370,7 @@ public static DataType commonType(DataType left, DataType right) {
                 return KEYWORD;
             }
             if (left == TEXT || right == TEXT) {
-                return TEXT;
+                return KEYWORD;
```

Comment:
I couldn't think of a case that this can cause problem as far as all the tests pass. I wonder if it can be simplified like below?

```
if (isString(left) && isString(right)) {
    return KEYWORD;
}
```

Reply:
You are right, I thought it would change the behaviour of `left == right`, but that is handled much earlier in this method. I will simplify.